manager2/src/app/user/user.service.ts

```diff
@@ -68,14 +67,15 @@ export class UserService {
             httpOptions)
     }
 
-    delete(userId: string) {
+    delete(userId: string, message: string) {
+        // console.log(userId, message);
         let httpOptions = {
             //headers: new HttpHeaders({
             //  'x-api-key': localStorage.getItem('my-api-key')
             //}),
         };
         return this.http.delete(
-            environment.apiUrl + '/user/' + userId ,
+            environment.apiUrl + '/user/' + userId + '/message/' + message,
```

Comment:
Why using a different url? Why not keeping current route and adding in delete body an optional message?
Furthermore adding message in request url is a bad idea (special chars, spaces, length...)

Reply:
I agree it is not so clean, but DELETE don't have a body, to add a body we have to use POST, I  don't know if using POST for DELETE is a good idea

Reply:
delete can have a body

Reply:
https://v7.angular.io/api/common/http/HttpClient#delete

Reply:
Maybe a Read the doc too quick :) I will try again ;)

Reply:
maybe httpclient does not provide option to do so, don't know, but protocol allows to do so (did it on several projects)
getting message as a request parameter is not good